composeApp/src/commonTest/kotlin/org/ooni/probe/data/repositories/PreferenceRepositoryTest.kt

```diff
@@ -0,0 +1,117 @@
+package org.ooni.probe.data.repositories
+
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.ooni.testing.createPreferenceDataStore
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class PreferenceRepositoryTest {
+    private lateinit var preferenceRepository: PreferenceRepository
+
+    @BeforeTest
+    fun before() {
+        preferenceRepository = PreferenceRepository(createPreferenceDataStore())
+    }
+
+    @AfterTest
+    fun after() =
+        runBlocking {
+            runTest {
+                preferenceRepository.clear()
+            }
+        }
+
+    @Test
+    fun testAllSettings() =
+        runBlocking {
```

Comment:
Usually `runBlocking` isn't needed, `runTest` is enough. Did you run into any trouble just with `runTest`?